generated/src/aws-cpp-sdk-s3-crt/source/model/CreateBucketRequest.cpp

```diff
@@ -34,6 +33,28 @@ CreateBucketRequest::CreateBucketRequest() :
 {
 }
 
+bool CreateBucketRequest::HasEmbeddedError(Aws::IOStream &body,
+  const Aws::Http::HeaderValueCollection &header) const
+{
+  // Header is unused
+  (void) header;
+
+  auto readPointer = body.tellg();
```

Comment:
not very clear why a similar logic is here and in ErrorMarshaller (well I guess I understand why, because one answers if there is such error and another parses and error to the object).

nit: would be nice to extract this logic into some common function in S3 Client to avoid copy-pasted generated code.

Reply:
While I agree, this cleanup can be done as part of a separate PR after further review of XML error parsing. The XML parser does a few more steps than this . As of now, we can keep S3 handling separate before redirecting to a common API